libs/math/include/math/matrix_operations.hpp

```diff
@@ -98,7 +98,7 @@ void Product(ArrayType const &obj1, typename ArrayType::Type &ret)
 template <typename ArrayType>
 void Sum(ArrayType const &obj1, typename ArrayType::Type &ret)
 {
-  ret = obj1.data().in_parallel().Reduce(memory::TrivialRange(0, obj1.size()),
+  ret = obj1.data().in_parallel().Reduce(memory::Range(0, obj1.size()),
                                          [](typename ArrayType::VectorRegisterType const &a,
                                             typename ArrayType::VectorRegisterType const &b) ->
                                          typename ArrayType::VectorRegisterType { return a + b; });
```

Comment:
Is it not possible to use `auto` here? Would make the code more readable.

Reply:
done